products/apigateway/api.yaml

```diff
@@ -0,0 +1,246 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Product
+name: ApiGateway
+versions:
+  - !ruby/object:Api::Product::Version
+    name: beta
+    base_url: https://apigateway.googleapis.com/v1beta/
+scopes:
+  - https://www.googleapis.com/auth/cloud-platform
+async: !ruby/object:Api::OpAsync
+  operation: !ruby/object:Api::OpAsync::Operation
+    path: 'name'
+    base_url: '{{op_id}}'
+    wait_ms: 1000
+  result: !ruby/object:Api::OpAsync::Result
+    path: 'response'
+    resource_inside_response: true
+  status: !ruby/object:Api::OpAsync::Status
+    path: 'done'
+    complete: True
+    allowed:
+      - True
+      - False
+  error: !ruby/object:Api::OpAsync::Error
+    path: 'error'
+    message: 'message'
+objects:
+  - !ruby/object:Api::Resource
+    name: 'Api'
+    min_version: beta
+    create_url:  projects/{{project}}/locations/global/apis?apiId={{name}}
+    self_link: projects/{{project}}/locations/global/apis/{{name}}
+    base_url:  projects/{{project}}/locations/global/apis
+    update_verb: :PATCH
+    update_mask: true
+    description: |
+      A consumable API that can be used by multiple Gateways.
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Official Documentation':
+          'https://cloud.google.com/api-gateway/docs/quickstart'
+      api: 'https://cloud.google.com/api-gateway/docs/reference/rest/v1beta/projects.locations.apis'
+    # parameters:
+    #   - !ruby/object:Api::Type::String
+    #     name: 'region'
+    #     description: |
+    #       The name of the Memcache region of the instance.
+    #     required: true
+    #     input: true
+    #     url_param_only: true
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'name'
+        description: |
+          The resource name of the API.
+        required: true
+        input: true
+        url_param_only: true
+        pattern: projects/{{project}}/locations/global/apis/{{name}}
+      - !ruby/object:Api::Type::String
+        name: 'displayName'
+        description: |
+          A user-visible name for the API.
+      - !ruby/object:Api::Type::String
+        name: 'managedService'
+        input: true
+        description: |
+          Immutable. The name of a Google Managed Service ( https://cloud.google.com/service-infrastructure/docs/glossary#managed).
+          If not specified, a new Service will automatically be created in the same project as this API.
+      # - !ruby/object:Api::Type::String
+      #   name: 'state'
+      #   description: |
+      #     The instance state - short description.
+      #   output: true
+      #   exclude: true
+      - !ruby/object:Api::Type::Time
```

Comment:
I don't think our users typically care about having createTime in state, so this can be removed (but it's not super important)

Reply:
^